src/parser/SparqlParserHelpers.cpp

```diff
@@ -4,45 +4,25 @@
 
 #include "SparqlParserHelpers.h"
 
-#include "../util/antlr/ThrowingErrorStrategy.h"
 #include "sparqlParser/generated/SparqlAutomaticLexer.h"
 
 namespace sparqlParserHelpers {
 using std::string;
 
-struct ParserAndVisitor {
- private:
-  string _input;
-  antlr4::ANTLRInputStream _stream{_input};
-  SparqlAutomaticLexer _lexer{&_stream};
-  antlr4::CommonTokenStream _tokens{&_lexer};
-
- public:
-  SparqlAutomaticParser _parser{&_tokens};
-  SparqlQleverVisitor _visitor;
-  explicit ParserAndVisitor(string input,
-                            SparqlQleverVisitor::PrefixMap prefixes)
-      : _input{std::move(input)}, _visitor{std::move(prefixes)} {
-    _parser.setErrorHandler(std::make_shared<ThrowingErrorStrategy>());
-  }
-
-  template <typename ResultType, typename ContextType>
-  auto parse(const std::string_view input, const std::string_view name,
-             ContextType* (SparqlAutomaticParser::*F)(void)) {
-    try {
-      auto context = (_parser.*F)();
-      auto resultOfParse =
-          std::move(context->accept(&(_visitor)).template as<ResultType>());
+ParserAndVisitor::ParserAndVisitor(string input)
+    : input_{std::move(input)}, visitor_{} {
```

Comment:
// The default in ANTLR is to log all errors to the console and to continue the parsing. We need to turn parse errors into exceptions instead to propagate them to the user.